api/src/opentrons/drivers/rpi_drivers/gpio.py

```
@@ -49,171 +55,121 @@
-    HALT pin high (normal running state), since the default value after export
-    is low.
-    """
-    for pin in sorted(OUTPUT_PINS.values()):
-        _enable_pin(pin, OUT)
-
-    for pin in sorted(INPUT_PINS.values()):
-        _enable_pin(pin, IN)
-
-
-def robot_startup_sequence():
-    """
-    Gets the robot ready for operation by initializing GPIO pins, resetting
-    the Smoothie and enabling the audio pin. This only needs to be done
-    after power cycling the machine.
-    """
-    initialize()
-
-    # audio-enable pin can stay HIGH always, unless there is noise coming
-    # from the amplifier, then we can set to LOW to disable the amplifier
-    set_high(OUTPUT_PINS['AUDIO_ENABLE'])
-
-    # smoothieware programming pins, must be in a known state (HIGH)
-    set_high(OUTPUT_PINS['HALT'])
-    set_high(OUTPUT_PINS['ISP'])
-    set_low(OUTPUT_PINS['RESET'])
-    time.sleep(0.25)
-    set_high(OUTPUT_PINS['RESET'])
-    time.sleep(0.25)
-
-
-def turn_on_blue_button_light():
-    set_button_light(blue=True)
-
-
-def set_button_light(red=False, green=False, blue=False):
-    color_pins = {
-        OUTPUT_PINS['RED_BUTTON']: red,
-        OUTPUT_PINS['GREEN_BUTTON']: green,
-        OUTPUT_PINS['BLUE_BUTTON']: blue
-    }
-    for pin, state in color_pins.items():
-        if state:
-            set_high(pin)
+MODULE_LOG = logging.getLogger(__name__)
+
+
+class GPIOCharDev:
+    def __init__(self, chip_name: str):
+        self._chip = gpiod.Chip(chip_name)
+        self._lines = self._initialize()
+
+    @property
+    def chip(self):
+        return self._chip
+
+    @property
+    def lines(self) -> Dict:
+        return self._lines
+
+    def _request_line(self, offset, name, request_type):
+        line = self.chip.get_line(offset)
+        try:
+            line.request(consumer=name, type=request_type, default_vals=[0])
+        except OSError:
+            MODULE_LOG.error(f'Line {offset} is busy. Cannot establish {name}')
```

we should not swallow this error, we should let this exception raise